lib/idle_task_queue.dart

```diff
@@ -0,0 +1,67 @@
+// @JS()
+library idle_task_queue;
+
+import 'dart:async';
+import 'dart:html';
+import 'dart:js';
+
+import 'package:meta/meta.dart';
+
+/// A handler which will be called during idle time.
+typedef void IdleTaskHandler<T>(T input);
+
+@immutable
+class _IdleTask<T> {
+  final IdleTaskHandler<T> handler;
+  final T input;
+
+  _IdleTask(this.handler, this.input);
+}
+
+/// Adds the task to a queue which will be called during idle time.
+void enqueueIdleTask<T>(IdleTaskHandler<T> handler, T input) {
+  // For unsupported browsers when the polyfill is not available,
+  // we still want the task to be called eventually.
+  if (!_doesBrowserSupportIdleCallback) {
+    new Future.delayed(const Duration()).then((_) => handler(input));
+    return;
+  }
+
+  _taskQueue.add(new _IdleTask<T>(handler, input));
+
+  // Start running the queue if it has not been started.
+  // If there is a task, the queue is already running and doesn't
+  // need to be started again.
+  if (_currentTask == null) {
+    _currentTask = window.requestIdleCallback(
+      allowInterop(_runQueue),
+      <String, dynamic>{'timeout': 1000},
+    );
+  }
+}
+
+void _runQueue(IdleDeadline deadline) {
+  while (!deadline.didTimeout &&
+      deadline.timeRemaining() > 0 &&
+      _taskQueue.isNotEmpty) {
+    final task = _taskQueue.removeAt(0);
+    task.handler(task.input);
```

Comment:
What happens if the task throws an unexpected exception or error here? Will it kill the queue?

Reply:
Good catch. That would probably break something as-is